src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+      ),
+      languageDetector = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        new LanguageDetector(LanguageDetectorAuth(siteSettings.translationSvcToken))
+      ),
+      sentimentDetectorAuth = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        SentimentDetectorAuth(siteSettings.translationSvcToken)
+      )
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided watchlist.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithWatchlist(langToWatchlist: Map[String, List[String]], delta: Delta = Delta()) = {
+    delta.copy(
+      langToWatchlist = Some(langToWatchlist)
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided blacklist.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithBlacklist(blacklist: List[BlacklistedTerm], delta: Delta = Delta()) = {
+    delta.copy(
+      blacklist = Some(blacklist)
+    )
+  }
+
+  private class MessageHandler extends IMessageHandler {
+    override def notifyException(exception: Throwable, phase: ExceptionPhase): Unit = {
+      logError("Service Bus client threw error while processing message.", exception)
+    }
+
+    /**
+      * Called by the QueueClient thread when a message arrives on the Service Bus.
+      *
+      * Note: this is configured in [[TransformManager.startQueueClient]] such that it's called serially (only 1 thread
+      *       will execute it at a time) which is necessary for the concurrency correctness of this module.
+      */
+    override def onMessageAsync(message: IMessage): CompletableFuture[Void] = {
+      // Wait for the previous update of transform context to settle
+      // to ensure we're calculating the delta against the latest context
+      writeLock.lock()
+      writeLock.unlock()
+
+      // Read the service bus message and build delta using data store.
+      val delta = Option(message.getProperties.getOrDefault("dirty", null)) match {
+        case Some(value) => value match {
+          case "settings" =>
+            val siteSettings = configManager.fetchSiteSettings()
+            deltaWithSettings(siteSettings)
+          case "watchlist" =>
+            val langToWatchlist = configManager.fetchWatchlist()
+            deltaWithWatchlist(langToWatchlist)
+          case "blacklist" =>
+            val blacklist = configManager.fetchBlacklist()
+            deltaWithBlacklist(blacklist)
+          case unknown =>
+            logError(s"Service Bus client received unexpected update request. Ignoring.: $unknown")
+            Delta()
+        }
+        case None =>
+          logError(s"Service Bus client received unexpected message. Ignoring.: ${message.toString}")
```

Comment:
Yes. For better or worse, I'd added it explicitly to allude to `message` not being a string.